docs/okeps/okep-5088-evpn.md

```diff
@@ -0,0 +1,789 @@
+into Provider Network 1. In this scenario, vm-2 is able to migrate into node-1 on the UDN network, preserving the same IP
+address it had in the external provider network. Similarly, there is another Provider Network 2 which may or may not
+correspond to another UDN within the Kubernetes cluster. However, notice that the red and blue networks are both using
+the same IP addressing scheme and sharing the same hardware, however due to VRF isolation they are completely unaware and
+unable to communicate with each other.
+
+### Using EVPN as the Overlay for User Defined Networks
+
+With integrating into a customer's already existing TOR spine and leaf architecture, Geneve can be disabled, and network
+segmentation will still persist for east/west traffic due to VXLAN tunnels with EVPN. This is true for both IP-VRFs,
+and MAC-VRFs. This reduces packet overhead for customers, while also providing some other advantages that come with
+EVPN, such as link redundancy and broadcast, unknown unicast, and multicast (BUM) traffic suppression.
+
+## Proposed Solution
+
+As mentioned in the previous sections, EVPN will continue to build upon the BGP support already implemented into
+OVN-Kubernetes using FRR. This support includes integration with an OVN-Kubernetes API as well as an FRR-K8S API for
+configuring BGP peering and routing for UDNs. FRR already supports EVPN and similar API resources will be leveraged to
+accomplish configuring FRR as the BGP/EVPN control plane. Recall that FRR relies on Linux netdevs to be configured in
+order for EVPN to work. There are two configuration modes in FRR to accomplish this.
+
+### Multiple VXLAN Devices (MVD)
+
+MVD is the classic way that FRR maps netdevs to EVPN configuration. In this model we create the following Linux constructs
+in order to use EVPN:
+
+1. A VRF device
+2. Linux Bridge enslaved to the VRF
+3. An SVI attached to the bridge (for IP-VRF)
+4. A VXLAN device enslaved to the bridge
+5. A VTEP IP configured locally (generally on a loopback interface)
+
+Devices 1-4 are needed to be configured for each network (UDN). Only a single VTEP IP is needed, although more than one
+could be configured if it was desired to use multiple tunnels (uncommon).
+
+### Single VXLAN Device (SVD)
+
+SVD is a newer way that was created in order to solve scale issues around creating too many netdevs with MVD. In this model
+only a single VXLAN device is created, along with a single Linux Bridge. VLANs are used within the Linux Bridge to segment
+networks. The following devices are required:
+
+1. A VRF device (one per UDN)
+2. Linux Bridge (single bridge with a VLAN per UDN)
+3. An SVI attached the bridge (for IP-VRF, one VLAN sub-interface per UDN)
+4. A VXLAN device enslaved to the bridge (added in the UDN VLAN)
+5. A VTEP IP configured locally (generally on a loopback interface)
+
+SVD is supposed to scale better than MVD, and is similar to how other physical routers allow a single VTEP to be used
```

Comment:
"scale better" how ? I understand it will use a single bridge w/ 1 VLAN + VLANed SVI per UDN instead of N bridges, but ... at the end of the day, this is limited by the number of VLANs. 

Since I have no idea on what's the upper boundary of the number of bridges, I'm not sure how much better this will scale.

Reply:
well we don't have the data, but I'm sure the FRR community would not have moved to SVD to solve the problem if MVD was better. There is some description here as to the scale problem:
https://github.com/FRRouting/frr/pull/12364